webrtc/videoroom/lib/videoroom/pipeline.ex

```diff
@@ -171,17 +213,60 @@ defmodule VideoRoom.Pipeline do
         end)
 
     spec = %ParentSpec{children: children, links: links}
-    state = update_in(state, [:tracks, track_id], &%Track{&1 | encoding: encoding})
+    endpoint = Endpoint.update_track(state.endpoints[endpoint_id], track_id, encoding)
+    state = put_in(state.endpoints[endpoint_id], endpoint)
     {{:ok, spec: spec}, state}
   end
 
-  def handle_notification({:signal, message}, {:endpoint, peer_pid}, _ctx, state) do
-    send(peer_pid, {:signal, message})
-    {:ok, state}
+  def handle_notification(
+        {:vad, val} = msg,
+        {:endpoint, endpoint_id} = from,
+        _ctx,
+        state
+      ) do
+    Membrane.Logger.info("#{inspect(msg)}, from: #{inspect(from)}")
+
+    {actions, display_managers} =
+      Enum.reduce(state.display_managers, {[], state.display_managers}, fn
+        {id, display_manager}, {actions, display_managers} = _acc when id != endpoint_id ->
+          case DisplayManager.update(display_manager, endpoint_id, val) do
+            {:ok, display_manager} ->
+              {actions, Map.put(display_managers, id, display_manager)}
+
+            {{:replace, old_id, new_id}, display_manager} ->
+              old_track_id =
+                Endpoint.get_video_tracks(state.endpoints[old_id])
+                |> List.first()
+                |> (fn %Track{id: id} -> id end).()
+
+              new_track_id =
+                Endpoint.get_video_tracks(state.endpoints[new_id])
+                |> List.first()
+                |> (fn %Track{id: id} -> id end).()
+
+              actions =
+                actions ++
+                  [
+                    {:forward, {{:endpoint, id}, {:disable_track, old_track_id}}},
+                    {:forward, {{:endpoint, id}, {:enable_track, new_track_id}}}
+                  ]
+
+              # send(endpoint_id, {:signal, {:replace_track, old_track_id, new_track_id}})
+              {actions, Map.put(display_managers, id, display_manager)}
+
+            {:error, :no_such_endpoint_id} ->
```

Comment:
shouldn't we fail here?

Reply:
I was thinking about it but raising here will kill the whole room

Reply:
Ok, but can this error ever happen?

Reply:
Acctually no so I removed it entirely